### ISSUE TITLE
dev.sh not working properly when using latest version of docker

### ISSUE THREAD

**Issue by sidtohan** [middlewarehq/middleware#383]
### Description:
Unable to run the dev.sh script. It gives an error in the docker-compose.yml file as well as an issue with the "watch" command.

### Steps to reproduce:

1. Start docker using docker desktop (currently using version 4.30.0-amd64)
2. Go to the root path of the repository and run dev.sh using
    `./dev.sh`
    
### Expected behavior:

dev.sh should start the application with the development environment.

### Actual behavior:

![image](https://github.com/middlewarehq/middleware/assets/82583557/74fa8d21-7921-49c8-9636-504f2cfb334f)

### Server Setup Information:

- Operating System: Pop!_OS 22.04 LTS
- Deployment Method: docker

### Additional context

I came across this discussion on [stack overflow.](https://stackoverflow.com/questions/66514436/difference-between-docker-compose-and-docker-compose) Maybe the current code uses an older version of docker compose?

Also, found this in the cli/source/app.tsx file:

![image](https://github.com/middlewarehq/middleware/assets/82583557/dcffa7a8-afa9-4ce4-b748-8c7891295278)

**Comment by jayantbh** [comment]
Hey @sidtohan! 👋🏽 
We'd be happy to help you with this.
Could you share the output of `docker version` with us?

Generally this issue was solved by simply asking people to update to the latest version of docker in the past.
But you're mentioning you're already on the latest version.

However, you also mentioned that you're on `Pop!_OS 22.04 LTS` which might simply not have a build for the version of docker that contains the `watch` command yet, making the latest version be not the actually latest release of docker.

For reference: The latest docker version listed online appears to be [24.0.9](https://docs.docker.com/engine/release-notes/24.0/#2407)

**Comment by sidtohan** [comment]
![image](https://github.com/middlewarehq/middleware/assets/82583557/993c2bad-b6e2-4ad3-8afa-cd56aac3ad02)

Here is the output for docker version command.

**Comment by adnanhashmi09** [comment]
Thanks @sidtohan for submitting the issue. The problem seems to be related to the version of docker installed. It is actually better if we move to using `docker compose` instead of `docker-compose` as the `docker-compose` project has probably stopped receiving updates. 

We could make the changes so that even older versions of docker and `docker-compose` is supported.

**Comment by adnanhashmi09** [comment]
I also think this is a good point to make some prerequisite checks before starting the container for development. This way we could provide better troubleshooting experience for the user. the prerequisite checks could include:
- Check if the correct version of docker is installed.
- Check if watch command is compatible
- Check if the docker-compose.yml file is present
- Check if ports specified are available to be used

**Comment by jayantbh** [comment]
Yea, a checklist that is part of the dev.sh script might work well to prevent similar issues.

Also, previously I had mentioned that the latest version is 24.0.7.
My bad, the latest is actually [26.1.3](https://docs.docker.com/engine/release-notes/26.1/#2613), and @sidtohan's version appears to be pretty close to that.

Perhaps as a fix to this at the moment, we should change the command from `docker-compose` to `docker compose` instead.

Think that'll work? @adnanhashmi09 @sidtohan

**Comment by adnanhashmi09** [comment]
> Yea, a checklist that is part of the dev.sh script might work well to prevent similar issues.
> 
> Also, previously I had mentioned that the latest version is 24.0.7. My bad, the latest is actually [26.1.3](https://docs.docker.com/engine/release-notes/26.1/#2613), and @sidtohan's version appears to be pretty close to that.
> 
> Perhaps as a fix to this at the moment, we should change the command from `docker-compose` to `docker compose` instead.
> 
> Think that'll work? @adnanhashmi09 @sidtohan

Yes that should work as a quick fix. 
But a better way would be to first try out `docker compose watch` and if it fails then try `docker-compose watch`. If both fail, we throw an error. For the sake of backwards compatibility.

**Comment by jayantbh** [comment]
@sidtohan just following up. Were you able to move past this?

**Comment by sidtohan** [comment]
Hey @jayantbh , still facing the same issue. I tried replacing the `docker-compose` commands in app.tsx with `docker compose`. However, I'm facing this issue now.

![image](https://github.com/middlewarehq/middleware/assets/82583557/57651255-2c07-4b9c-8fd6-ec3abfbb12a8)

**Comment by adnanhashmi09** [comment]
@sidtohan  What changes did you make? I don't think you are using the spawn command correctly. Please refer to the nodejs docs. For some reference you are supposed to use the `runCommand` function like this:
```js
const commandResponse = runCommand(
  'docker',
  ['compose', 'watch'],
  runCommandOpts
);
```

**Comment by sidtohan** [comment]
> @sidtohan What changes did you make? I don't think you are using the spawn command correctly. Please refer to the nodejs docs. For some reference you are supposed to use the `runCommand` function like this:
> 
> ```js
> const commandResponse = runCommand(
>   'docker',
>   ['compose', 'watch'],
>   runCommandOpts
> );
> ```

Yes, you are right @adnanhashmi09, my bad. I have used it properly now, and the error doesn't occur anymore.
![image](https://github.com/middlewarehq/middleware/assets/82583557/be1f64a8-8f2c-40dc-b303-e68c65f5d140)



On that note, can this issue be assigned to me? I would like to implement the proper fix as suggested above(running `docker compose` first and then reverting to `docker-compose` should it fail)

**Comment by samad-yar-khan** [comment]
@sidtohan done!

**Comment by adnanhashmi09** [comment]
Closing this, as this was fixed in #431
